mk/docker.mk

```diff
@@ -3,11 +3,18 @@ KUMA_VERSION ?= master
 
 DOCKER_SERVER ?= docker.io
 DOCKER_REGISTRY ?= $(DOCKER_SERVER)/kumahq
+DOCKER_DEV_IMAGE_SUFFIX ?=
 DOCKER_USERNAME ?=
 DOCKER_API_KEY ?=
 
+ifdef IMAGES_REPOSITORY_SUFFIX
+ifneq ($(strip $(IMAGES_REPOSITORY_SUFFIX)),)
+DOCKER_DEV_IMAGE_SUFFIX = -$(IMAGES_REPOSITORY_SUFFIX)
+endif
+endif
+
 define build_image
-$(addsuffix :$(BUILD_INFO_VERSION)$(if $(2),-$(2)),$(addprefix $(DOCKER_REGISTRY)/,$(1)))
+$(addsuffix :$(BUILD_INFO_VERSION)$(if $(2),-$(2)),$(addprefix $(DOCKER_REGISTRY)/,$(1)$(DOCKER_DEV_IMAGE_SUFFIX)))
```

Comment:
Can't be actually just do:

```
$(DOCKER_REGISTRY)/$(if $(2),$(2)-)$(1)$(if $(DOCKER_DEV_IMAGE_SUFFIX),-$(DOCKER_DEV_IMAGES_SUFFIX)):$(BUILD_INFO_VERSION)
```

Seems like it's equivalent and more readable to me no?

Reply:
Also any specific rational for prefix vs suffix?

Reply:
I feel suffix is more used in different repos:
https://hub.docker.com/r/kong/kong-gateway-dev
https://hub.docker.com/r/kong/kong-gateway-dev
https://hub.docker.com/r/envoyproxy/envoy-dev